Dev sum2020

jolin0504 opened the pull request:
request review on Inference for Proportions

jolin0504 committed:
First Final version

jolin0504 committed:
Reference modified

jolin0504 committed:
some change in the 'acknowledgement'

jolin0504 committed:
last updated 0708

jolin0504 committed:
last update 07/13

neilhatfield reviewed (changes requested):
Here are the things that need to be changed:

1. Add `library(boastUtils)` to ui.R file.
2. Add App Meta Data to the top of the ui.R file, just below the library calls
3. Organize your code:
    - Rather than having this:
        ```
        dashboardPage(skin="purple",
              
              #Title
              dashboardHeader(title="Inference for Proportions",titleWidth=250,
                              
                              tags$li(class = "dropdown",
                                      tags$a(href='https://shinyapps.science.psu.edu/',
                                             icon("home")))
                              ),
              
              #Sidebar
              dashboardSidebar(
                width = 250,
    - Strive for this:
        ```
        dashboardPage(
          skin="purple",
          #Title
          dashboardHeader(
            title="Inference for Proportions",
            titleWidth=250,
            tags$li(class = "dropdown",
                    tags$a(href='https://shinyapps.science.psu.edu/',
                           icon("home")))
          ),
          #Sidebar
          dashboardSidebar(
            width = 250,
4. Remove any extra empty lines in your code; you can have one between different blocks of code.
5. Resolve any long lines in your code. Reformatting your code as listed above will help, but there are still many lines of code that are over 80 characters long.
6. In the UI,
    - If we don't have a Z score table, delete the menu item (line 24), don't just comment it out.
    - Proportions needs to be capitalized in line 43
    - Change "This app visualizes" to "This app shows" (you could also use "displays", your choice). in line 44.
    - The second instruction has a strange " - for the confidence interval app -". This needs to be fixed as this makes it sound like the user has to go to some other app.
    - The third and fourth "instructions" aren't instructions. They are just explaining the other pages in the app. Instructions are what the user should *do*, not explanations of the tabs.
    - We need to come up with another word than "Design", which shouldn't have a colon (:) after it. My suggestion is to use the word "Context".
    - Make the UP Residency Percentage (fix the misspelling) layout and the Difference of Proportions lay match; particularly in reference to the context information.
    - All plots are missing alt text; this MUST be added.
    - Fix the font sizes in the plots for the Z* score tab
    - What is supposed to go in the Feedback area of this tab?
    - Z* needs to be styled consistently throughout the app.

I've stopped this review as I need to code to be reformatted to be more readable. My preference is to NOT use `titlePanel`, `sidebarLayout`, `sidebarPanel`, and `mainPanel`; rather, use `fluidRow` and `column` so that we have a bit more control over the layout. Additionally, you should not have a well panel inside of another well panel (which you currently have at line 155 (`sidebarPanel` invokes a well). In general, commented code should only be left for a particular reason that is documented. Otherwise, commented code should be removed. This helps to ensure that commented lines are actually comments.

jolin0504 committed:
latest version for 'Inference for Porportions'

neilhatfield reviewed (changes requested):
Make the following changes:

1. Resolve Issue #2; remember, the goal is to have the app run as silently as possible
2. Replace "z*" with `\\(Z^*\\)` everywhere.
3. In the UI,
    - Lines 61-69 should be
        ```
        p("This app....in 2020.",
          br(),
          br(),
          br(),
          div(class = "updated", "Last Update:...")
        )
        ```
    - Remove the `tags$em` from Context
 
Review is stopped. Please be sure to address ALL of my comments from the last time as well as this time before resubmitting. This includes the my comments about `fluidPage`, `titlePanel` etc. You also need to be explicit in your code.

jolin0504 committed:
Latest version

jolin0504 committed:
latest latest version of this app

jolin0504 committed:
formatted

jolin0504 committed:
updated

jolin0504 committed:
update explore button

neilhatfield committed:
Convert to app.R

I've converted to the app.R framework. You'll need to use this version going forward.

neilhatfield reviewed (changes requested):
This is going to be a short review because many of the things that I requested in the first review still have not been done. You will need to pull the code from GitHub before you make any changes as I have converted everything to the app.R framework.

1. Bring all of the code into alignment with the Style Guide. This includes but is not limited to:
    a. Formatting the code to maximize readability (see the first review for an example)
    b. Explicitly name the arguments for any function that is taking more than one
    c. Look through the code I posted for "FIX THIS" these are comments of things that need fixing
    d. Fix long lines in your code. Rstudio has a tool which can help: Click on the Tools menu and select Global Options from the bottom of the menu. Click on "Code" in the left-hand menu, then the Display tab across the top. The fourth option down should be "Show margin"; check this box. Immediately below that is Margin column; make sure that box shows 80. This will make a grey vertical line appear in your code that will alert you to long lines (any line that goes beyond). Keep in mind that somethings such as URLs should not be broken and will need to go beyond the margin.
2. You MUST resolve Issue #2 as I'm still getting the message in the console.
3. I'm not sure that we need to duplicate the Finding Z* page in this app (it is already in the Inference for Means app). If it is to remain, you need to make it match what is in the Means app.
4. Font sizes for all graphs need to be increased
5. We might need to consider combining the Proportion and Sample bar charts into one with side-by-side bars to improve spacing.
6. We need to re-think the layout for the Difference of Proportions page. Specifically, the first well panel containing Population and Sample Info shouldn't be in a well panel and pushes the controls really far down the page. Come up with a suggestion of a new layout.

We're about 2.5 weeks from needing to use this app.

jolin0504 committed:
most recently update 11/3/2020

neilhatfield committed:
Update app.R

Major changes to bring into alignment with the Inference for Means app as well as addressing other issues.

neilhatfield reviewed (approved):
Approving my edits of the previous work
